src/cpu/arm.rs

```diff
@@ -12,6 +12,11 @@
 // OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
 // CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
+#![cfg_attr(
+    not(any(target_arch = "aarch64", target_arch = "arm")),
+    allow(dead_code)
+)]
```

Comment:
This doesn't seem right to me. Why is it here?

Reply:
There is a similar line at the top of `src/cpu/intel.rs` [here](https://github.com/briansmith/ring/blob/main/src/cpu/intel.rs#L15). Without it, building on ppc64le yields dead code warnings, e.g.:
```
warning: method `available` is never used
   --> src/cpu/arm.rs:206:12
    |
204 | impl Feature {
    | ------------ method in this implementation
205 |     #[inline(always)]
206 |     pub fn available(&self, _: super::Features) -> bool {
    |            ^^^^^^^^^
```

These warnings will probably appear on any non-ARM or non-Intel platform. They don't appear on Intel because they are referenced in functions like `detect_implementation()` in [`src/aead/aes.rs`](https://github.com/briansmith/ring/blob/main/src/aead/aes.rs#L404) -- however in this example, default `_cpu_features` are used on any non-Intel or non-ARM platform, and skip the calls to the cpu-specific functions.